backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/reader/impl/IndexReaderProvider.java

```diff
@@ -12,18 +12,11 @@
 
 public interface IndexReaderProvider {
 
-	/**
-	 * Closes and drops any cached resources (index readers in particular).
-	 * <p>
-	 * Should be used when stopping the index or to clean up upon error.
-	 */
-	void clear() throws IOException;
-
 	/**
 	 * Make sure the index reader returned by the next call to {@link #getOrCreate()}
 	 * will return an up-to-date view of the index.
 	 */
-	void refresh();
+	void refresh() throws IOException;
```

Comment:
Considering what `IndexReaderProvider#refresh` does and how it's used, it should probably be called `clear`. Yeah I know, this was my suggestion, but it was wrong :x 

Also, for the javadoc:

```
	/**
	 * Closes and drops any cached resources (index readers in particular).
	 * <p>
	 * Should be used when stopping the index, to clean up upon error, or simply to force the creation of a new reader (refresh) on the next call to {@link #getOrCreate()}.
	 */
```